include/envoy/network/address.h

```diff
@@ -161,6 +162,11 @@ class Instance {
    * @return the type of address.
    */
   virtual Type type() const PURE;
+
+  /**
+   * @return address ttl.
+   */
+  virtual std::chrono::seconds ttl() const PURE;
```

Comment:
I'm not sure that address should have a TTL associated with it at the interface level. Can we just store the TTL inside the strict/logical DNS cluster and use it to reset the timer when configured? I think you can just add TTL to the DNS resolve interface.

Reply:
I was considering add TTL as another parameter 

https://github.com/envoyproxy/envoy/blob/034b23e956e2381105c605e0d851a7ea6c0ca849/include/envoy/network/dns.h#L41

In this case, I am going to add another type 
```c++
  typedef std::function<void(const std::list<Address::InstanceConstSharedPtr>&& address_list, const std::chrono::seconds& ttl)>
      ResolveCbWithTtl;
```

Does this change look better?

Reply:
Yup, or just change the existing callback to include TTL. The caller can ignore if they don't care. That is probably even better.

Reply:
Sorry, it's not clear to me. Do you mean without adding another type, we change the current `ResolveCb` to 
```c++
typedef std::function<void(const std::list<Address::InstanceConstSharedPtr>&& address_list, const std::list<std::chrono::seconds>&& ttl)> ResolveCb;
```

Reply:
Change the resolve callback to take TTLs. I would probably change it pass a list of structs where each struct includes an address and a TTL.